conjure-runtime/src/per_host_clients.rs

```diff
@@ -0,0 +1,67 @@
+//! Per-host-client types.
+use std::collections::{hash_map, HashMap};
+
+use url::Url;
+
+/// A collection of clients for each host of a replicated service.
+pub struct PerHostClients<T> {
+    pub(crate) clients: HashMap<Host, T>,
+}
+
+impl<T> PartialEq for PerHostClients<T> {
+    fn eq(&self, other: &Self) -> bool {
+        self.clients.len() == other.clients.len()
+            && self.clients.keys().all(|k| other.clients.contains_key(k))
+    }
+}
+
+impl<T> PerHostClients<T> {
```

Comment:
The API here is very minimal currently - I expect we'll probably expand it in the future but this is enough to support basic workflows.